changelog_unreleased/typescript/15811.md

```diff
@@ -0,0 +1,24 @@
+#### Improve conditional type alias layout (#15811 by @seiyab)
+
+<!-- prettier-ignore -->
+```ts
+// Input
+type FallbackFlags<F extends Flags | undefined> =
+  Equals<NonNullableFlag<F>["flags"], {}> extends true
+    ? Dict<any>
+    : NonNullableFlag<F>["flags"];
+
+// Prettier stable
+type FallbackFlags<F extends Flags | undefined> =
+  Equals<NonNullableFlag<F>["flags"], {}> extends true
+    ? Dict<any>
+    : NonNullableFlag<F>["flags"];
+
+// Prettier main
+type FallbackFlags<F extends Flags | undefined> = Equals<
+  NonNullableFlag<F>["flags"],
+  {}
+> extends true
+  ? Dict<any>
+  : NonNullableFlag<F>["flags"];
+```
```

Comment:
stable/main output should be swapped?

Reply:
Yes, thanks. I was confused.
Can we replace the term "main" into something like "next" in template? I first wrote them correctly and mistakenly swapped them "Oh, main branch should be ugly compared to this branch." without seeing the term "stable". 😓

Reply:
It used to be "master", it reference to "main" branch.

Reply:
Makes sense.